src/configure-webpack/build-pages-config.spec.js

```diff
@@ -65,43 +65,129 @@ describe('pages webpack preset', function () {
 
-      const commons = webpackConfig.plugins.filter((plugin) => plugin instanceof optimize.CommonsChunkPlugin)
-      expect(commons[0].minChunks).equal(2)
+        const html = webpackConfig.plugins.filter((plugin) => plugin instanceof HtmlWebpackPlugin)
+        expect(html.length).equal(2)
+
+        const firstOptions = html[0].options
+        expect(firstOptions.chunks).deep.eql([ 'vendor', 'index' ])
+        expect(firstOptions.filename).deep.eql('index.html')
+        expect(firstOptions.template).deep.eql('index.html')
+
+        const secondOptions = html[1].options
+        expect(secondOptions.chunks).deep.eql([ 'vendor', 'demo' ])
+        expect(secondOptions.filename).deep.eql('demo.html')
+        expect(secondOptions.template).deep.eql('demo.html')
+      })
+
+      it('should have the CommonsChunkPlugin enabled', function () {
+        const webpackConfig = buildPagesConfig(['index', 'demo'], { vendor: true }, baseConfig)
+
+        const vendor = webpackConfig.plugins.filter((plugin) => plugin instanceof optimize.CommonsChunkPlugin)
+        expect(vendor.length).equal(1)
+      })
+
+      it('should set name to "vendor" in the CommonsChunkPlugin', function () {
+        const webpackConfig = buildPagesConfig(['index', 'demo'], { vendor: true }, baseConfig)
+
+        const vendor = webpackConfig.plugins.filter((plugin) => plugin instanceof optimize.CommonsChunkPlugin)
+        expect(vendor[0].chunkNames[0]).equal('vendor')
+      })
+
+      it('should set minChunks to a function in the CommonsChunkPlugin', function () {
+        const webpackConfig = buildPagesConfig(['index', 'demo'], { vendor: true }, baseConfig)
+
+        const vendor = webpackConfig.plugins.filter((plugin) => plugin instanceof optimize.CommonsChunkPlugin)
+        expect(vendor[0].minChunks).to.be.a('function')
+      })
+
+      it('should set pages as chunks in the CommonsChunkPlugin', function () {
+        const webpackConfig = buildPagesConfig(['index', 'demo'], { vendor: true }, baseConfig)
+
+        const vendor = webpackConfig.plugins.filter((plugin) => plugin instanceof optimize.CommonsChunkPlugin)
+        expect(vendor[0].selectedChunks[0]).equal('index')
+        expect(vendor[0].selectedChunks[2]).equal('demo')
+      })
+    })
+
+    describe('when a page is flagged as independent', function () {
+      it('should be excluded from both the vendor and common chunks', function () {
+        const webpackConfig = buildPagesConfig([
+          'index',
+          { name: 'demo', independent: true }
+        ], {
+          vendor: true,
+          common: true
+        }, baseConfig)
+
+        const plugins = webpackConfig.plugins.filter((plugin) => plugin instanceof optimize.CommonsChunkPlugin)
+        const vendor = plugins[0]
+        const common = plugins[1]
+
+        expect(vendor.selectedChunks[0]).equal('index')
+        expect(vendor.selectedChunks[2]).equal(undefined)
```

Comment:
Maybe test the length of the `selectedChunks` here to be equal 1?

Reply:
For some reason there's an `undefined` on every second position in the array, so it will actually be 2. Will check that.